nion/swift/ExportDialog.py

```diff
@@ -263,20 +263,38 @@ class UnitType(enum.Enum):
 class ExportSizeModel(Observable.Observable):
     def __init__(self, display_item: DisplayItem.DisplayItem) -> None:
         super().__init__()
-        display_size = self.__calculate_display_size_in_pixels(display_item)
-        self.__width = display_size.width
-        self.__height = display_size.height
-        self.__aspect_ratio = self.__width / self.__height
-        self.__units = UnitType.PIXELS
-        self.__float_to_string_converter = Converter.FloatToStringConverter()
-        self.__primary_field = 'width'  # Primary field to determine which text is calculated
-        self.__enforce_width_height_constraints()
+        if display_item.data_item is not None:
```

Comment:
Generally, putting initializers in `__init__` inside an `if` statement is prone to error since it is impossible for other methods to know whether the class has initialized fully or not. If you were to do something like this, you could just do the check and assign `self.__image_2d` either a value or `None` if it wasn't available. However, there is no need to store a field `self.__image_2d` - it can just be removed and calculated on the fly.